doc/Requirements.md

```diff
@@ -0,0 +1,374 @@
+Configuration are then applied when their associated "scenario" is detected.
+
+"Scenario" are detected through arbitrary criterion provided by the PF host.
+
+## Definition
+
+Configurations: set of values for different parameters.
+A configuration MUST not contain 2 values of the same parameters.
+For example, given a PF with 3 integer parameters A,B,C,
+a configuration can contain
+ - 1 value for (A), (B) or (C)
+ - 2 value for (A,B), (A,C) or (B,C)
+ - 3 value for (A,B,C)
+
+## Configuration
+
+A PF MUST offer configurations as describe in the Definition chapter.
+(Note: rule based parameter engine does not manipulate directly values,
+      it applies configuration on the parameters.)
+(Why: This is what the reference implementation does.)
+
+Each configuration MUST be associated with a predicate that condition its
+eligibility. A configuration with a true predicate is called an "eligible configuration"
+(Why: This is what the reference implementation does.)
+
+The predicate MUST be a "selection criterino rule". See next chapter for a definition.
+
+## Selection criterion
+
+A selection criterion MUST have one, and only one, state at a given time.
+
+A selection criterion MUST know at construction all possible states.
+(Why: To be able to validate:
+       - rules on start
+       - state changes)
+
+The selection criterion possible states MUST be specifiable by:
+ - Directly a state set (Input -> states == identity)
+   (Note: called exclusive criterion)
+   (Why: any criterion can be crated with this one.)
+
+The selection criterion possible states SHOULD be specifiable by:
+ - A combination of values
+   (Note: combination in the mathematical sense “ab” -> [“”, “a”, “b”, “ab”])
+   (Note: called inclusive criterion)
+   (Why: The reference implementation supports it.)
+
+### Definitions
+ - Selection criterion rule: function (in the mathematical sense) that MUST given selection criteria return a Boolean.
+
+ - Rule: an Boolean expression of Selection criterion rules
+   (Note: implementation only allows AND and OR combination)
+
+### Criterion change
+
+The api to change criterion values MUST allow atomicity regarding configuration application.
+Ie. It MUST be possible to change multiple criterion values without triggering a configuration
+application.
+(Why: Two criterion might have an excluding state.
+      If configuration application was triggered after each )
+
+### Rules
+
+One MUST always be able to create a selection criterion rule from a given selection criterion
+(Why: if no rules can be formulated from a criterion, it is useless)
+
+Parameter values change SHOULD be selected by Rules.
+(Why: A rule based inference engine has been chosen based on implementation and configuration ease)
+
+## Domains
+
+### Definition
+Domain: ordered set of configuration containing the values for the same parameters.
+
+### Requirement
+
+Each configuration SHOULD be in a "domain" (see Definition chapter).
```

Comment:
> Domains are mostly a way to define the priority of "Scenarios" for some
parameters. It is not a MUST because this goal could also be achieve with (for
example) global configurations and per parameter priority. It is not a MAY
because the reference implementation uses domains.
:+1